src/main/kotlin/com/openlattice/data/storage/PostgresDataQueries.kt

```diff
@@ -58,51 +65,44 @@ fun buildPreparableFiltersSql(
         metadataOptions: Set<MetadataOption>,
         linking: Boolean,
         idsPresent: Boolean,
-        partitionsPresent: Boolean
+        partitionsPresent: Boolean,
+        detailed: Boolean = false
 ): Pair<String, Set<SqlBinder>> {
     val filtersClauses = buildPreparableFiltersClause(startIndex, propertyTypes, propertyTypeFilters)
     val filtersClause = if (filtersClauses.first.isNotEmpty()) " AND ${filtersClauses.first} " else ""
-
     val metadataOptionColumns = metadataOptions.associateWith(::mapMetaDataToColumnSql)
-    val nonAggregatedMetadataSql = metadataOptionColumns
-            .filter { !isMetaDataAggregated(it.key) }.values.joinToString("")
-    val innerGroupBy = groupBy(ESID_EKID_PART_PTID + nonAggregatedMetadataSql)
-    // TODO: remove IS NOT NULL post-migration
-    val linkingClause = if (linking) {
-        " AND ${ORIGIN_ID.name} IS NOT NULL AND ${ORIGIN_ID.name} != '${IdConstants.EMPTY_ORIGIN_ID.id}' "
-    } else {
-        ""
-    }
+    val metadataOptionColumnsSql = metadataOptionColumns.values.joinToString("")
+    val innerGroupBy = groupBy(ESID_EKID_PART_PTID)
+    val outerGroupBy = if (metadataOptions.contains(MetadataOption.ORIGIN_IDS)) {
```

Comment:
If ENTITY_KEY_IDS metadata option is set, it should be either in this group by or aggregated in the outer query.
It is used for materialized views btw, to collect the origin ids for linking entities. Which will probably need another discussion because of the re-creation of linked entities shouldn't be possible, but otherwise neighbor queries are not possible in materialized views.

Reply:
As much as I would love to fix all the issues, this wasn't implemented previously (at least not correctly) and reading linked entities is currently not implemented correctly either. You can see these in the code to read linked entities which has to re-shuffle all the data for every single read call in memory after read each component entry individually. 

I have aggregated origin_ids in and aliased the column in the out call, but I'm not sure this will fix the complete problem for materialized views.